### PR TITLE
refactor(cluster): improve scylla.yaml patching

### PR DESCRIPTION
**Pull request by enaydanov**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [x] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [x] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [x] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Review by fruch — APPROVED** [review]
I liked the idea, simplify it alot

File: sdcm/cluster.py [hunk 1]
```diff
@@ -1699,178 +1700,151 @@ def datacenter_setup(self, datacenters):
...
-                scylla_yml[key] = value
-
-        scylla_yaml_contents = yaml.safe_dump(scylla_yml)
-        with open(yaml_dst_path, 'w') as scylla_yaml_file:
-            scylla_yaml_file.write(scylla_yaml_contents)
-        self.log.debug("Scylla YAML configuration:\n%s", scylla_yaml_contents)
-        wait.wait_for(self.remoter.send_files, step=10, text='Waiting for copying scylla.yaml to node',
-                      timeout=300, throw_exc=True, src=yaml_dst_path, dst='/tmp/scylla.yaml')
-        self.remoter.run('sudo mv /tmp/scylla.yaml {}'.format(yaml_file))
+    def update_scylla_yaml(self, path=SCYLLA_YAML_PATH):
```
Comment (bentsi) [inline sdcm/cluster.py:1703]: `update_scylla_yaml` - > `scylla_yaml_remote`| `remote_scylla_yaml`

**Review by bentsi — CHANGES_REQUESTED** [review]
please address small comment

File: sdcm/cluster.py (hunk 1 above)
Comment (enaydanov) [reply sdcm/cluster.py]: done

**Review by bentsi — APPROVED** [review]

**Pull request merged**